azerothcore/azerothcore-wotlk
# fix(DB/SAI): Void Baron Galaxis encounter.

## Changes Proposed:
- Added the void cloud (It's a visual creature that is within "summoned old god" and it's the only place where it's being used, the movement modification has no effect on any interactions regarding that creature/quest, tested to be sure). (Quest id is 10808)
- Fixed item having cast time
- fixed movement
- Fix drain life to heal 1k per tick
- Fix npc going to spawn position on reset
- Slightly improve the add spawning (it's 0.5s faster)

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14453
- Closes https://github.com/chromiecraft/chromiecraft/issues/4706

## SOURCE:

https://www.youtube.com/watch?v=Wx3J4Wvnd48
## Tests Performed:

- tested

## How to Test the Changes:


.q a 10295
.go xyz -1235 1356 5.3 530
Use the item, notice it doesn't have a cast time.
Look for the void storm cloud at spawn location.
.damage 5000
wait for drain life spell to be casted, confirm that it heals 1k per tick 

Reset the fight (.gm on), confirm that he stays on the ground and doesn't go back to where he was spawned from.

Can also manually test the healing by:
.cast 34236
.cast 34239

## Known Issues and TODO List:


- [ ] Void Storm Cloud (visual) creature is only used at one more place, at Summoned Old God, for quest 10808. I've double checked and modification on the movement template have no effect there. (It's stationary visual creature, that was already on the ground)


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
